test/abstractarray.jl

```diff
@@ -1280,3 +1280,15 @@ end
     @test Int[0 t...; t... 0] == [0 1 2; 1 2 0]
     @test_throws ArgumentError Int[t...; 3 4 5]
 end
+
+@testset "issue #39896, modified getindex " begin
+    a=ones(20000)
+    ax=axes(a,1)
+    @test a==a[ax]
+    ax=UnitRange(1,10)
+    @test a[ax]==ones(10)
+    a=ones(BigInt(20000))
+    ax=UnitRange(1,20000)
+    @test a[ax]== ones(BigInt(20000))
+
+end
```

Comment:
This should have tests for `Base.OneTo` with endpoints smaller than the length of the array as well. I think it would also be good to have tests for higher dimensional arrays to make sure that too works as expected. You could use a for loop to test with these different combinations of arrays and ranges.

It is also not necessary to use sizes of 20000 here, as that will just increase the time and memory it takes to run these tests unnecessarily. Unless you are explicitly testing something for large arrays, working with sizes around 10 or so is fine.

Reply:
Changes made:
- OneTo for smaller endpoints added.
- Tests for 2-dimensional arrays added.
- Now array of 10 are only used.

Reply:
> OneTo for smaller endpoints added.

I don't see this anywhere. Note that `1:x` is not the same as `Base.OneTo(x)`.

Reply:
Used the Base.OneTo for smaller endpoints now. @simeonschaub